### ISSUE TITLE
Generated Network fails to load in viewer or editor. IndexOutOfBounds in NetworkLine.setEdges

### ISSUE THREAD

**Issue by gt670dn** [OpenMetroMaps/OpenMetroMaps#8]
After manually resolving #6 and #7 the next problem on loading the network in the Viewer or the Editor:
```
Exception in thread "main" de.topobyte.utilities.apache.commons.cli.commands.RunnerException: java.lang.IndexOutOfBounds
Exception: Index: 1, Size: 1
        at de.topobyte.utilities.apache.commons.cli.commands.ExeRunner.run(ExeRunner.java:91)
        at de.topobyte.utilities.apache.commons.cli.commands.ExeRunner.run(ExeRunner.java:44)
        at org.openmetromaps.cli.OpenMetroMapsCli.main(OpenMetroMapsCli.java:115)
Caused by: java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
        at java.util.ArrayList.rangeCheck(Unknown Source)
        at java.util.ArrayList.get(Unknown Source)
        at org.openmetromaps.maps.graph.NetworkLine.setEdges(NetworkLine.java:45)
        at org.openmetromaps.maps.graph.LineNetworkBuilder.addAllEdges(LineNetworkBuilder.java:124)
        at org.openmetromaps.maps.graph.LineNetworkBuilder.addLines(LineNetworkBuilder.java:99)
        at org.openmetromaps.maps.graph.LineNetworkBuilder.<init>(LineNetworkBuilder.java:60)
        at org.openmetromaps.maps.ModelUtil.ensureView(ModelUtil.java:139)
        at org.openmetromaps.maps.viewer.MapViewer.init(MapViewer.java:291)
        at org.openmetromaps.maps.viewer.MapViewer.<init>(MapViewer.java:144)
        at org.openmetromaps.cli.maps.RunMapViewer.main(RunMapViewer.java:73)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at de.topobyte.utilities.apache.commons.cli.commands.ExeRunner.run(ExeRunner.java:84)
```
Here is the network file: [kvv.txt](https://github.com/OpenMetroMaps/OpenMetroMaps/files/1745763/kvv.txt) (As txt because Github does not support xml-files)

Same problem occurs with the VRS (Verkehrsverbund Rhein-Sieg, Cologne/Bonn), but I have a feeling that one would be too large anyway to display in one map.

[vrs.txt](https://github.com/OpenMetroMaps/OpenMetroMaps/files/1745786/vrs.txt)
[GTFS VRS](https://transitfeeds.com/p/verkehrsverbund-rhein-sieg/835/latest/download)

**Comment by sebkur** [comment]
Yes, there were problems with lines consisting of just two stops / one segment.
108 (RVS), 136 (AVG) and 138 (AVG) in KVV. 9278f5fa fixes this.

**Comment by sebkur** [comment]
Still, I can't load the resulting file into the editor, it just hangs

**Comment by gt670dn** [comment]
Same here for me. It stops at "Durlach Turmberg" during load.

**Comment by sebkur** [comment]
I just deleted everything except for the S-Bahn lines and then I can load the file

**Comment by sebkur** [comment]
#FFFFFF is not a good default color for lines without color specified via GTFS though

**Comment by sebkur** [comment]
Not sure whether the file has just too many lines or if there's some line in particular that causes a problem. Feels like an infinite loop

**Comment by sebkur** [comment]
After removing lines 21 (VBK), 246 (FMO) and 248 (FMO) manually from the file, it loads

**Comment by gt670dn** [comment]
Works for me. It's not that you can actually read that clumped-together map, but the import is successful. ;) 

Could it be because of Line 21 (VBK) stopping two times at Durlach Turmberg?

**Comment by sebkur** [comment]
Yes, that must be it. 248 (FMO) has stops occurring multiple times, too. 246 (FMO) is not relevant, removing 21 (VBK) and 248 (FMO) suffices

**Comment by gt670dn** [comment]
If I define Line 21 (VBK) as circular = true it works.

**Comment by sebkur** [comment]
Interesting. I think a minimal file that causes this problem will be vital for finding the root problem

**Comment by sebkur** [comment]
Since the original problem of this issue is solved, I'm closing this, to be continued in #9
